Library/Workflow/ManagerWorker.php

```diff
@@ -36,29 +36,40 @@ protected function usage()
      * Init work with parameters
      *
      * @return void
+     *
+     * @throws WorkflowException
+     * @throws \Exception
      */
     protected function initWorker()
     {
         $this->getDI()->get('logr')->debug(json_encode(func_get_args()));
         parent::initWorker();
-        $serviceFullName = $this->workflowConfig['servicesClassNameSpace'] . "\\" .
-            ucFirst($this->workflowConfig['name']) . 'Manager';
-        // xxx Allow client to overwrite ?
-        $this->workflowManagerService = new ManagerService($this->paramHash['workflow']);
+        $this->workflowManagerService = new ManagerService($this->workflowConfigDto);
+        $serviceFullName = $this->getServiceFullName();
         $this->getDI()->get('logr')->debug('Loading ' . $serviceFullName);
         $this->service = new $serviceFullName();
 
-        $this->topicName = Topic\TopicService::getWorkflowManagerTopicName($this->workflowConfig['name']);
+        $this->topicName = Topic\TopicService::getWorkflowManagerTopicName($this->workflowConfigDto->getWorkflowName());
+    }
+
+    /**
+     * Get service full name
+     *
+     * @return string
+     */
+    private function getServiceFullName() : string
+    {
+        return $this->workflowConfigDto->getServicesClassNameSpace() . "\\" . ucFirst($this->workflowConfigDto->getWorkflowName()) .
```

Comment:
Maybe it could be safer to check the Class name validity, at least by checking dto func returns or even more by checking the class exists